controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -156,16 +159,18 @@ public void deleteStream(StreamInfo request, StreamObserver<DeleteStreamStatus>
 
         log.info(requestTag.getRequestId(), "deleteStream called for stream {}/{}.",
                 request.getScope(), request.getStream());
-        authenticateExecuteAndProcessResults(() -> this.authHelper.checkAuthorization(request.getScope() + "/" +
-                        request.getStream(), AuthHandler.Permissions.READ_UPDATE),
+        authenticateExecuteAndProcessResults(() -> this.authHelper.checkAuthorization(
+                AuthResourceRepresentation.ofAStreamInScope(request.getScope(), request.getStream()),
```

Comment:
This one of the semantic changes proposed here, that delete depends on the permission over scope and stream right?

Reply:
Yes, the delete depends on the permission over the respective resource. That's not a semantic change though, as the behavior was the same in the pre-PR code, as well. I had modified it to use parent resource in this PR but then changed it back based on our auth model discussions in the respective issue.